api/metric/api.go

```diff
@@ -18,64 +18,169 @@ import (
 	"context"
 
 	"go.opentelemetry.io/api/core"
-	"go.opentelemetry.io/api/registry"
 	"go.opentelemetry.io/api/unit"
 )
 
-type MetricType int
+// Kind categorizes different kinds of metric.
+type Kind int
 
+//go:generate stringer -type=Kind
 const (
-	Invalid    MetricType = iota
-	Gauge                 // Supports Set()
-	Cumulative            // Supports Inc()
+	Invalid        Kind = iota
+	CumulativeKind      // Supports Add()
+	GaugeKind           // Supports Set()
+	MeasureKind         // Supports Record()
 )
 
+// Recorder is the implementation-level interface to Set/Add/Record individual metrics.
+type Recorder interface {
+	// Record allows the SDK to observe a single metric event
+	Record(ctx context.Context, value float64)
+}
+
+// LabelSet represents a []core.KeyValue for use as pre-defined labels
+// in the metrics API.
+//
+// TODO this belongs outside the metrics API, in some sense, but that
+// might create a dependency.  Putting this here means we can't re-use
+// a LabelSet between metrics and tracing, even when they are the same
+// SDK.
+type LabelSet interface {
+	Meter() Meter
+}
+
+// Meter is an interface to the metrics portion of the OpenTelemetry SDK.
 type Meter interface {
-	// TODO more Metric types
-	GetFloat64Gauge(ctx context.Context, gauge *Float64GaugeHandle, labels ...core.KeyValue) Float64Gauge
+	// DefineLabels returns a reference to a set of labels that
```

Comment:
I meant that you cannot read the labels from a LabelSet. I don't think you would need to, but the point is that the SDK is not required to keep this information, it could potentially stream it and forget--retain only an identifier.